test/test_io.py

```diff
@@ -190,50 +192,50 @@ def test_read_timestamps_pts_unit_sec(self):
             num_frames = int(round(float(stream.average_rate * stream.time_base * stream.duration)))
             expected_pts = [i * pts_step * stream.time_base for i in range(num_frames)]
 
-            self.assertEqual(pts, expected_pts)
             container.close()
+            assert pts == expected_pts
 
-    def test_read_partial_video_pts_unit_sec(self):
+    @pytest.mark.parametrize('start', range(5))
+    @pytest.mark.parametrize('offset', range(1, 4))
+    def test_read_partial_video_pts_unit_sec(self, start, offset):
         with temp_video(10, 300, 300, 5, lossless=True) as (f_name, data):
             pts, _ = io.read_video_timestamps(f_name, pts_unit='sec')
 
-            for start in range(5):
-                for offset in range(1, 4):
-                    lv, _, _ = io.read_video(f_name, pts[start], pts[start + offset - 1], pts_unit='sec')
-                    s_data = data[start:(start + offset)]
-                    self.assertEqual(len(lv), offset)
-                    assert_equal(s_data, lv)
+            lv, _, _ = io.read_video(f_name, pts[start], pts[start + offset - 1], pts_unit='sec')
+            s_data = data[start:(start + offset)]
+            assert len(lv) == offset
+            assert_equal(s_data, lv)
 
             container = av.open(f_name)
             stream = container.streams[0]
             lv, _, _ = io.read_video(f_name,
                                      int(pts[4] * (1.0 / stream.time_base) + 1) * stream.time_base, pts[7],
                                      pts_unit='sec')
+            container.close()
```

Comment:
If we are touching this here, lets do it directly in the preferred way: instead of doing

```python
container = av.open(...)
...
container.close()
```

lets do

````python
with av.open(...) as container:
    ...
```

This applies to all occurrences of this pattern.